### PR TITLE
Homework number 11

### PR DESCRIPTION
**Pull request by Bloodstinger**
The explanation in the respective README files, if there any.

### CODE REVIEW

File: src/lecture/ten/getCallerClassAndMethodName/GetCallerClassAndMethodName.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package lecture.ten.getCallerClassAndMethodName;
+
+public class GetCallerClassAndMethodName {
+    public static String getCallerClassAndMethodName() {
+        StackTraceElement[] stackTraceElements = new Throwable().getStackTrace();
+        try {
+            String result = stackTraceElements[2].getClassName() + "#" + stackTraceElements[2].getMethodName();
```
Comment (Sofasmile) [inline src/lecture/ten/getCallerClassAndMethodName/GetCallerClassAndMethodName.java]: we can simplify:
```suggestion
            return stackTraceElements[2].getClassName() + "#" + stackTraceElements[2].getMethodName();
```

File: src/lecture/ten/simmetricDifference/SymmetricDifference.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package lecture.ten.simmetricDifference;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class SymmetricDifference {
+    public static <T> Set<T> symmetricDifference(Set<? extends T> set1, Set<? extends T> set2) {
+        Set commonness = new HashSet();
```
Comment (Sofasmile) [inline src/lecture/ten/simmetricDifference/SymmetricDifference.java]: we can simplify this method:
```suggestion
        Set commonness = new HashSet(set1);
```

File: src/lecture/ten/numberSequence/Main.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
+package lecture.ten.numberSequence;
+
+import java.util.ArrayDeque;
+import java.util.Iterator;
+import java.util.Scanner;
+
+public class Main {
+    public static void main(String[] args) {
+        int i = 0;
+        Integer container;
```
Comment (Sofasmile) [inline src/lecture/ten/numberSequence/Main.java:10]: maybe better move "Integer container" in the while cycle

Comment (Bloodstinger) [reply src/lecture/ten/numberSequence/Main.java:10]: This container is used in two different while cycles. If I move it to one of them, the other won't see it. I'd prefer just to use it as a global variable, is that okay?

File: src/lecture/ten/simmetricDifference/SymmetricDifference.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+public class SymmetricDifference {
+    public static <T> Set<T> symmetricDifference(Set<? extends T> set1, Set<? extends T> set2) {
+        Set commonness = new HashSet(set1);
+        Set difference = new HashSet(set2);
+        commonness.retainAll(difference);
+        difference.addAll(set1);
+        difference.addAll(set2);
+        difference.removeAll(commonness);
+        return difference;
+
```
Comment (Sofasmile) [inline src/lecture/ten/simmetricDifference/SymmetricDifference.java]: remove redundant line

File: src/lecture/ten/getCallerClassAndMethodName/GetCallerClassAndMethodName.java [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+public class GetCallerClassAndMethodName {
+    public static String getCallerClassAndMethodName() {
+        StackTraceElement[] stackTraceElements = new Throwable().getStackTrace();
+        try {
+            return stackTraceElements[2].getClassName() + "#" + stackTraceElements[2].getMethodName();
+        } catch (ArrayIndexOutOfBoundsException e) {
+            return null;
+        }
+    }
+
```
Comment (Sofasmile) [inline src/lecture/ten/getCallerClassAndMethodName/GetCallerClassAndMethodName.java]: remove redundant line

File: src/lecture/ten/getCallerClassAndMethodName/Test.java [hunk 6]
```diff
@@ -0,0 +1,11 @@
+package lecture.ten.getCallerClassAndMethodName;
+
+public class Test {
+    public static void main(String[] args) {
+        System.out.println(GetCallerClassAndMethodName.getCallerClassAndMethodName());
+        test();
+    }
```
Comment (Sofasmile) [inline src/lecture/ten/getCallerClassAndMethodName/Test.java:7]: add a blank line after first method

**Review by Sofasmile — APPROVED** [review]
remember about CS

**Pull request merged**